model/task/task.go

```diff
@@ -1213,6 +1213,9 @@ func (t *Task) GetDisplayStatus() string {
 	if t.DisplayStatus != "" {
 		return t.DisplayStatus
 	}
+	if !t.IsFinished() {
```

Comment:
I'm confused about why this would help. Wouldn't something that's not finished get all the way to 1237? Or does the ticket you're referring to have some of the detail metadata below?

Reply:
The latter. The task was in a status of undispatched but had end details. Weird data aside, I felt that adding this check was a reasonable validation check, but I can see it maybe being a bit too much

Reply:
I think if this happens again we'll have more data to solve the underlying cause which might be better? Plus undispatched is still the wrong status right?

Reply:
If it's reproducible yes, but I think this change is still worth making. Without this the status would show setup-failed which is worse

Reply:
Ah I see, I was in the legacy UI where it only says aborted, but yeah setup-failed is worse.